src/Microsoft.AspNetCore.Routing/Template/TemplateBinder.cs

```diff
@@ -56,14 +57,20 @@ public class TemplateBinder
         /// <param name="pattern">The <see cref="RoutePattern"/> to bind values to.</param>
         /// <param name="defaults">The default values for <paramref name="pattern"/>. Optional.</param>
         /// <param name="requiredKeys">Keys used to determine if the ambient values apply. Optional.</param>
-        /// <param name="parameterPolicyFactory">The <see cref="ParameterPolicyFactory"/>.</param>
+        /// <param name="constraints">
+        /// A list of (<see cref="string"/>, <see cref="IRouteConstraint"/>) pairs to evalute when producing a URI.
+        /// </param>
+        /// <param name="parameterTransformers">
+        /// A list of (<see cref="string"/>, <see cref="IOutboundParameterTransformer"/>) pairs to evalute when producing a URI.
+        /// </param>
         public TemplateBinder(
             UrlEncoder urlEncoder,
             ObjectPool<UriBuildingContext> pool,
             RoutePattern pattern,
             RouteValueDictionary defaults,
             IEnumerable<string> requiredKeys,
-            ParameterPolicyFactory parameterPolicyFactory)
+            IEnumerable<(string parameterName, IRouteConstraint constraint)> constraints,
+            IEnumerable<(string parameterName, IOutboundParameterTransformer transformer)> parameterTransformers)
```

Comment:
What do you think about TemplateBinder ctor taking a enumerable of parameterName + IParameterPolicy. Dividing the the policies into constraints and transformers would happen inside the template binder. That way if something new is added to policies in the future we don't need to change the public signature.

Reply:
Solid!